src/config.rs

```diff
@@ -902,10 +902,12 @@ impl Config {
             .map(|p| p.as_path())
     }
 
-    pub fn shell(&self) -> Option<&Path> {
+    pub fn shell(&self) -> Option<&str> {
         self.shell
             .as_ref()
             .map(PathBuf::as_path)
+            .map(Path::to_str)
```

Comment:
We can get rid of this nasty conversion if `shell` is a `String` instead of a `PathBuf`.